test/dynamo/test_subclasses.py

```diff
@@ -676,6 +675,31 @@ def test_basic_autograd(self):
     def test_basic_autograd_inductor(self):
         self._test_autograd("inductor")
 
+    def test_unbind(self):
+        nt, _ = self._get_jagged_tensor(((2, 3, 4), 3), None)
+        nt2, _ = self._get_jagged_tensor(((2, 3, 5), 2), None)
+        nt3, _ = self._get_jagged_tensor(((2, 3, 4, 5), 3), None)
+
+        def fn(x):
+            return x.unbind()
+
+        compiled_f = torch.compile(fn, fullgraph=True, backend="eager", dynamic=True)
+        out = compiled_f(nt)
+
+        out_ref = fn(nt)
+
+        # correctness
+        self.assertEqual(len(out), len(out_ref))
+        for x, x_ref in zip(out, out_ref):
+            self.assertTrue(torch.allclose(x, x_ref))
+
+        # We specialize on the length of offsets, e.g. (1) we recompile if the
```

Comment:
Makes sense that if the user calls `unbind()`, we specialize on number of components.

However, for the SAM use case, I don't think this is a full solution and it'd be nice to avoid recompiling if the batch size changes. Right now, we're using an unbind-based fallback to do jagged broadcasting for jagged + dense, but I think we should move away from unbind() and tuck away the broadcasting logic behind a custom op or something.

Reply:
Yeah having a custom sounds good to avoid recompilation if there aren't too many